src/sickle.ts

```diff
@@ -33,6 +33,8 @@ export let closureExternsBlacklist: string[] = [
   'exports',
   'global',
   'module',
+  'WorkerGlobalScope',
```

Comment:
Not clear to me - are you suggesting I do that reorganization in this PR,
or is this a followup issue?

On Thu, Mar 17, 2016 at 3:17 PM Rado Kirov <notifications@github.com> wrote:

> In src/sickle.ts
> <https://github.com/angular/sickle/pull/103#discussion_r56586924>:
>
> > @@ -33,6 +33,8 @@ export let closureExternsBlacklist: string[] = [
> >    'exports',
> >    'global',
> >    'module',
> > +  'WorkerGlobalScope',
>
> This list needs to be more structured (we can keep hacking on it for now).
> Here is what I did in clutz:
>
>    -
>
>    fix which files are considered "platform apis" on both sides:
>    - lib.d.ts for TS (TSsymbols)
>       - closure/externs:COMMON (ClosureSymbols)
>    -
>
>    modify lib.d.ts to be a strict subset of closure externs
>    - enumerate symbols in missingPlatformSymbols = ClosureSymbols \
>    TSSymbols ( \ is set difference)
>
> clutz now emits everything not in closure externs + missingPlatformSymbols.
>
> Beyond "platform externs", I don't think the tool should be doing any
> white/black listing, as we can't have any visibility into what those
> symbols are. It won't scale to keep adding them here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/sickle/pull/103/files/eb39d1a2693cdb2848e5c5ea3135ea4c53cc04cb#r56586924>
>